install.sh

```diff
@@ -71,6 +71,45 @@ if (($SUPPORTS_SSE42 == 0)); then
     exit 1
 fi
 
+replace_tsdb() {
+    if (
+        [ -f "$SENTRY_CONFIG_PY" ] &&
+        ! grep -xq 'SENTRY_TSDB = "sentry.tsdb.redissnuba.RedisSnubaTSDB"' "$SENTRY_CONFIG_PY"
+    ); then
+        tsdb_settings="SENTRY_TSDB = \"sentry.tsdb.redissnuba.RedisSnubaTSDB\"
+
+# Automatic switchover 90 days after $(date). Can be removed afterwards.
+SENTRY_TSDB_OPTIONS = {\"switchover_timestamp\": $(date +%s) + (90 * 24 * 3600)}"
+
+        if grep -q 'SENTRY_TSDB_OPTIONS = ' "$SENTRY_CONFIG_PY"; then
+            echo "Not attempting automatic TSDB migration due to presence of SENTRY_TSDB_OPTIONS"
+        else
+            echo "Attempting to automatically migrate to new TSDB"
+            # Escape newlines for sed
+            tsdb_settings="${tsdb_settings//$'\n'/\\n}"
+            cp "$SENTRY_CONFIG_PY" "$SENTRY_CONFIG_PY.bak"
+            sed -i -e "s/^SENTRY_TSDB = .*$/${tsdb_settings}/g" "$SENTRY_CONFIG_PY" || true
+
+            if grep -xq 'SENTRY_TSDB = "sentry.tsdb.redissnuba.RedisSnubaTSDB"' "$SENTRY_CONFIG_PY"; then
+                echo "Migrated TSDB to Snuba. Old configuration file backed up to $SENTRY_CONFIG_PY.bak"
+                return
+            fi
+
+            echo "Failed to automatically migrate TSDB. Reverting."
+            mv "$SENTRY_CONFIG_PY.bak" "$SENTRY_CONFIG_PY"
+        fi
+
+        echo "FAIL: Your Sentry configuration uses a legacy data store for time-series data. Remove the options SENTRY_TSDB and SENTRY_TSDB_OPTIONS from $SENTRY_CONFIG_PY and add:"
```

Comment:
Still not sure if this script is the right place to fail. I'd say once we remove this TSDB, we can error out in Sentry like other things?

Reply:
The error message in Sentry will probably only appear in docker logs, right? This does not seem like a userfriendly way to inform them their config is broken.

One of the goals of this PR Is to unblock removal of Redis TSDB

Reply:
>The error message in Sentry will probably only appear in docker logs, right? This does not seem like a userfriendly way to inform them their config is broken.

Yes and I agree that it is not the best way to surface. That being said, for this to be consistent, we'd need a full config audit in the install script. I'd still go with a warning and not failing the script completely.

Reply:
I actually don't know based on which condition I would let Sentry fail. This error is only triggered if the automatic migration failed which I deem unlikely. I would like to encourage users to migrate asap as any delay (incurred by the user ignoring warnings) extends the timeframe for which we have to keep the redis tsdb around. We do have to keep the redis tsdb for at least 90 days (prob much more) because migrating redis data over to snuba takes that long for users.

So to me the outcome of not erroring in the install script would seem to be:

* users who need to manually migrate will overlook the warning
* when we start erroring in sentry eventually, the error will be invisible because it will occur in a docker container

Reply:
>I actually don't know based on which condition I would let Sentry fail. This error is only triggered if the automatic migration failed which I deem unlikely.

I concur. It seems unlikely for someone to hit this case.

>I would like to encourage users to migrate asap

Don't we already by automatically migrating them when possible?

> as any delay (incurred by the user ignoring warnings) extends the timeframe for which we have to keep the redis tsdb around. We do have to keep the redis tsdb for at least 90 days (prob much more) because migrating redis data over to snuba takes that long for users.

Not necessarily. Soon we'll have named releases which we can point people to, if they need to migrate or stay on Redis. Even without named releases, we can point them to a specific revision where we still have the RedisTSDB code.

>users who need to manually migrate will overlook the warning

I'd argue that if you have modified the TSDB setting and need a manual migration, you would be a person unlikely to look past a warning or even use this script :)

>when we start erroring in sentry eventually, the error will be invisible because it will occur in a docker container

I thought the container would simply refuse to start due to a missing class, forcing people to look into Docker logs, am I mistaken?

---

In any case, if you feel strongly about keeping failing in the install script, I'm fine by reserving my right to remove this in the future if it becomes a problem. I also trust you to monitor the forums for complaints around this.

Reply:
I don't really want to monitor the forum so I just removed the `exit 1`. I still think this will bite us in the ass once we want to remove redistsdb